### ISSUE TITLE
1.5.2 prouduces more hallucination than ever before

### ISSUE THREAD

**Issue by AeneasZhu** [ggerganov/whisper.cpp#1709]
After upgrading to ver 1.5.2, whisper.cpp produces more repetive hallucination than ever before. Here is the example:

<img width="868" alt="image" src="https://github.com/ggerganov/whisper.cpp/assets/44701446/2150dc4b-4ab7-4675-861f-8bd3729fa72c">

<img width="847" alt="image" src="https://github.com/ggerganov/whisper.cpp/assets/44701446/5d21d3d9-e07a-4fed-a52c-8071eae7420b">

While it works much better for ver 1.5.1. Hope it could be fixed in the next update.

Info:
OS: Win11
Whisper.cpp: 1.5.2
Whisper model: Large-v2

**Comment by AeneasZhu** [comment]
https://www.nationalreview.com/podcasts/capital-record/capital-record-year-in-review/

The tested audio link.

**Comment by mrfragger** [comment]
I only used 1.5.1 and just installed 1.5.3 and had no problems with ggml large and medium up to 12 mins of that mp3. Here is some of the output from the medium Core ML model.

```
[00:00:00.000 --> 00:00:03.600]   [Music]
[00:00:03.600 --> 00:00:11.840]   Politics, music, technology, roller coasters, golf carts, and the greatest country on earth.
[00:00:11.840 --> 00:00:18.000]   National Review's new show, the Charles C. W. Cook podcast, that's me, explores the scenic highways
[00:00:18.000 --> 00:00:23.360]   and byways of American political and cultural life. Featuring interviews with leading writers,
[00:00:23.360 --> 00:00:28.240]   thinkers, and public figures, every episode offers a fresh perspective on the promises and
[00:00:28.240 --> 00:00:34.080]   challenges facing America. Don't miss out. Tune into the Charles C. W. Cook podcast,
[00:00:34.080 --> 00:00:39.520]   on Apple Podcasts, Spotify, or wherever you get your shows.
[00:00:39.520 --> 00:00:50.480]   [Music]
[00:00:50.480 --> 00:00:59.600]   Hello and welcome to the very final Capital Record of 2023. I am your host David Bonson
[00:00:59.600 --> 00:01:06.640]   and I am excited to bring the year to a close here at National Review's Capital Record and to do so
[00:01:06.640 --> 00:01:13.840]   with a short but sweet trip down memory lane, kind of recapping some of the various guests
[00:01:13.840 --> 00:01:21.680]   and speakers and conversations we've enjoyed throughout the year. In looking back over this,
[00:01:21.680 --> 00:01:30.480]   you know, we're now wrapping up our third year of Capital Record. I am really enjoying doing this.
[00:01:30.480 --> 00:01:38.880]   I've frankly, more than anything else, enjoyed just the conversations I get to have each week
[00:01:38.880 --> 00:01:45.920]   with the various guests. The conversations for me are not really a podcast recording.
[00:01:45.920 --> 00:01:53.040]   They're me talking to someone that I learn from, that I enjoy about a conversation topic that I
[00:01:53.040 --> 00:02:01.840]   enjoy, and to the extent that we ever pull it off in a way that it edifies or informs you, then
[00:02:01.840 --> 00:02:06.480]   that's an added bonus. Obviously, that's why you would be listening, is if you're getting something
[00:02:06.480 --> 00:02:15.120]   out of it. But really, you know, selfishly, I just have learned and grown and enjoyed so much
[00:02:15.120 --> 00:02:22.160]   for each of these last three years, the weekly conversations I have that result in our Thursday
[00:02:22.160 --> 00:02:28.960]   posting of a Capital Record podcast. We fully intend for there to be a fourth year and I look
[00:02:28.960 --> 00:02:38.800]   forward to what 2024 will hold. Let's go back a little bit and look at what some of 2023 had to
[00:02:38.800 --> 00:02:46.480]   offer. It's funny because I, in looking over this list as I was preparing to record, there's a kind
[00:02:46.480 --> 00:02:52.240]   of trip down memory lane where I'll remember not only a given week and that recording and that
[00:02:52.240 --> 00:02:57.920]   topic, but then certain things that were happening in the news cycle at the time or in the markets in
[00:02:57.920 --> 00:03:05.920]   which I work. And part of it is surreal that some of this stuff took place, you know, so long ago
[00:03:05.920 --> 00:03:13.440]   because I remember many of these things as if they were yesterday. But we're at the end of 2023.
[00:03:13.440 --> 00:03:19.920]   It's been a very surprising year for a lot of people in the markets, meaning financial markets.
[00:03:19.920 --> 00:03:26.480]   We talk about that here on Capital Record. If we're going to do a podcast in defense
[00:03:26.480 --> 00:03:34.080]   of capital markets and their usefulness as a tool in a free enterprise system, then there is room
[00:03:34.080 --> 00:03:42.160]   to talk about stock markets, bond markets, currency, housing, monetary policy, a lot of
[00:03:42.160 --> 00:03:52.160]   these things that I think are not at the heart of defense of a free market, the defense of a free
[00:03:52.160 --> 00:03:59.760]   society, but are themselves directly related to the subject. And in some cases, they're tools
[00:03:59.760 --> 00:04:06.320]   with which these things can be enhanced. That's how I've always felt about capital markets. I
[00:04:06.320 --> 00:04:12.480]   think that they are a means, not an end, but they're a vital means, a means worth defending
[00:04:12.480 --> 00:04:19.040]   because the end is worth defending. And if that end of human flourishing is facilitated
[00:04:19.840 --> 00:04:26.400]   and enhanced, and in some cases at the speed of light, and in some cases with dramatic magnitude
[00:04:26.400 --> 00:04:35.280]   by things like robust financial markets, then somebody has to go defend them. And those people
[00:04:35.280 --> 00:04:42.800]   who, albeit sometimes are well-meaning, but ignorantly so, who will say they want to defend
[00:04:42.800 --> 00:04:50.480]   the cause of human flourishing or the end destination of human flourishing, but that they do not believe
[00:04:50.480 --> 00:05:00.400]   financial markets are necessary in facilitating such, I think that we have to counter that position
[00:05:00.400 --> 00:05:09.920]   with an intelligent defense of what capital markets have meant to market economy and what they
[00:05:10.560 --> 00:05:21.040]   must mean going forward. And so I've always been pretty self-aware and tried to be very transparent
[00:05:21.040 --> 00:05:27.600]   about what our agenda is. The agenda for me is not to defend Wall Street. It is, however,
[00:05:27.600 --> 00:05:38.000]   to defend Wall Street's role in what I'm really after, which is that flourishing that comes
[00:05:38.800 --> 00:05:45.440]   through a greater material and spiritual well-being is the byproduct of a good life,
[00:05:45.440 --> 00:05:52.320]   and that good life we are after has both material and immaterial components to it.
[00:05:52.320 --> 00:05:57.840]   That's what this podcast is about. That's pretty much what I've devoted my life to.
[00:05:57.840 --> 00:06:05.360]   When I go to work every day and pursue financial solutions on behalf of clients who pay us
[00:06:06.240 --> 00:06:13.760]   to give them advice and to solve problems and to be a solutions provider, that's what I'm doing.
[00:06:13.760 --> 00:06:21.840]   I'm harnessing the capital markets to be a tool. And within capital markets,
[00:06:21.840 --> 00:06:29.440]   this is the beauty of how all of this so virtuously and cyclically works together,
[00:06:30.560 --> 00:06:37.920]   pro-cyclically works together, that you have people who are human beings with certain goals and needs
[00:06:37.920 --> 00:06:44.160]   and wants and aspirations, and that their goals and needs and wants and aspirations can be met
[00:06:44.160 --> 00:06:55.680]   with other people performing their goals, needs, wants, aspirations. That by Mr. Smith, who has a
[00:06:55.680 --> 00:07:04.080]   certain financial return and need, let's say in cash flow income need, he can receive the dividend
[00:07:04.080 --> 00:07:11.840]   of ABC Public Company, and ABC Public Company can only meet the needs of Mr. Smith if they're out
[00:07:11.840 --> 00:07:16.800]   selling goods and services to other people that have needs and wants, that their products, their
[00:07:16.800 --> 00:07:23.760]   goods, their services meet. And so we harness free enterprise towards other aims within free enterprise.
[00:07:24.640 --> 00:07:30.080]   And I believe it's beautiful. And I think it's majestic. And I think it's provided the world
[00:07:30.080 --> 00:07:38.000]   its greatest advancement in quality of life at a speed at which we never would have thought
[00:07:38.000 --> 00:07:45.440]   possible. And I think that we basically do it invisibly. We do it almost entirely without any
[00:07:45.440 --> 00:07:52.400]   self-consciousness. And so I want to record a podcast for 30 to 60 minutes a week to try to
```

**Comment by ggerganov** [comment]
@AeneasZhu Could you check if the issues are resolved for you with v1.5.3?

**Comment by ggerganov** [comment]
How do you build `whisper.cpp` and what command do you use?

**Comment by AeneasZhu** [comment]
> How do you build `whisper.cpp` and what command do you use?

I directly download the built asset--[whisper-cublas-11.8.0-bin-x64.zip]. I have few knowledge on build.  However, I succeeded in building the ver 1.5.0 following the instruction: https://github.com/ggerganov/whisper.cpp/issues/878

**Comment by AeneasZhu** [comment]
> How do you build `whisper.cpp` and what command do you use?

<img width="853" alt="image" src="https://github.com/ggerganov/whisper.cpp/assets/44701446/727cbd0c-89a4-4d46-9427-26d5239d01a8">

I build it myself following the example:

```
cmake -B build -D WHISPER_CUBLAS=1
cmake --build build -j --config Release
```

It goes wrong as before.

**Comment by AeneasZhu** [comment]
It seems the issue has been fixed upon the 1.5.4 update. Thanks.
